File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
from collections import deque

def solution(numbers, target):
    answer = 0
    
    queue = deque()
    queue.append((0,0))
    
    while queue:
        index, value = queue.popleft()
        if index == len(numbers) and value == target:
            answer += 1
        elif index == len(numbers):
            pass
        else:
            queue.append((index+1, value + numbers[index]))
            queue.append((index+1, value - numbers[index]))

    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.swift
import Foundation

func solution(_ numbers: [Int], _ target: Int) -> Int {
    return dfs(numbers, target, 0, 0)
}

func dfs(_ numbers: [Int], _ target: Int, _ index: Int, _ sum: Int) -> Int {
    if index == numbers.count {
        return sum == target ? 1 : 0
    }
    
    let add = dfs(numbers, target, index + 1, sum + numbers[index])
    let subtract = dfs(numbers, target, index + 1, sum - numbers[index])
    
    return add + subtract
}